### ISSUE TITLE
Create Kanbans out of any objects

### ISSUE THREAD

**Issue by Bonapara** [twentyhq/twenty#2483]
# Desired behavior - Views

With this new field type, users can create `Pipeline views` on any object. Here's how:

### 1. Click on "Create a view"

![CleanShot 2023-11-13 at 16 34 58](https://github.com/twentyhq/twenty/assets/19412894/83fdc654-1a45-406e-a86f-dd7b8cb3607f)

### 2. Define the view parameters

![CleanShot 2023-11-13 at 16 38 18](https://github.com/twentyhq/twenty/assets/19412894/bee4dc3a-1c67-4210-8d1d-70668ac89be5)

1. Select a icon & a name for the view
2. Select the "Pipeline" view type

![CleanShot 2023-11-13 at 16 42 40](https://github.com/twentyhq/twenty/assets/19412894/7b748360-6470-4f89-a5f4-a14b70547003)

3. After selecting the pipeline view type, specify the Select field to be used as pipeline steps in the `Stages` dropdown.

![CleanShot 2023-11-13 at 16 50 15](https://github.com/twentyhq/twenty/assets/19412894/d9bc08ae-85af-4049-be97-e822aa38fe2a)

4. Create the new view

![CleanShot 2023-11-13 at 16 48 20](https://github.com/twentyhq/twenty/assets/19412894/b44287fa-01b9-42cc-a87e-e29fed1d7a0e)

5.If no `Select field type` are defined on the object, a deactivated dropdown is displayed. The main call-to-action (CTA) redirects to the `New field` page in the settings for the corresponding object. For example, for the company, go to `Settings > Company > New field`.

# Figmas

https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?type=design&node-id=15014-58618&mode=design&t=BPGkFQDJNvsZsRl7-11

```[tasklist]
### Tasks
- [ ] https://github.com/twentyhq/twenty/issues/2820
```

**Comment by ARSHIL1804** [comment]
@Bonapara  I would like to work on this. Can you guide me on the next task after @thaisguigon has created the Status component?

**Comment by Bonapara** [comment]
Hi @ARSHIL1804! Thank you for offering to contribute! I'll let @thaisguigon decide if we need external contributors for any part of this. Otherwise I added some new "Good first issue" you can check.

**Comment by ARSHIL1804** [comment]
Oh, I have made significant progress on this one. Should I raise a pull request for you to review and decide?

**Comment by ARSHIL1804** [comment]
Btw I have used the component developed by @thaisguigon in this #2820.

**Comment by Bonapara** [comment]
@thaisguigon WDYT? @ARSHIL1804 I'm not developing or merging PR ;)

**Comment by ARSHIL1804** [comment]
@thaisguigon  Please review this PR. If you are okay give me heads up so I can continue with Views

**Comment by charlesBochet** [comment]
Technical input:
- Add kanbanFieldMetadataId on view standard object
- build a component for view picker that should update the view / create a new view

**Comment by charlesBochet** [comment]
We can close this one, it has been released as part of v0.3.3!
